### PR TITLE
DSCForLinux

### PR DESCRIPTION
**Pull request by balukambala**
-Updated DSC, OMI packages.
-Added additional parameters in the Register operation.
-Updated documentation accordingly.

### CODE REVIEW

File: DSC/README.md [hunk 1]
```diff
@@ -25,7 +25,12 @@ Here're all the supported public configuration parameters:
 
 * `FileUri`: (optional, string) the uri of the MOF file/Meta MOF file/custom resource ZIP file.
 * `ResourceName`: (optional, string) the name of the custom resource module
-* `Mode`: (optional, string) the functional mode, valid values: Push, Pull, Install, Remove, Register. If not specified, it's considered as Push mode by default.
+* `Operation`: (optional, string) the functional operation, valid values: Register, Push, Pull, Install, Remove. If not specified, it's considered as Push Operation by default.
+* `NodeConfigurationName`: (optional, string) the name of the configuration to apply.
+* `RefreshFrequencyMins`: (optional, int) pecifies how often (in minutes) DSC attempts to obtain the configuration from the pull server.
```
Comment (eshaparmar) [inline DSC/README.md]: Minor: missing an S from specifies

File: DSC/README.md [hunk 2]
```diff
@@ -25,7 +25,12 @@ Here're all the supported public configuration parameters:
 
 * `FileUri`: (optional, string) the uri of the MOF file/Meta MOF file/custom resource ZIP file.
 * `ResourceName`: (optional, string) the name of the custom resource module
-* `Mode`: (optional, string) the functional mode, valid values: Push, Pull, Install, Remove, Register. If not specified, it's considered as Push mode by default.
+* `Operation`: (optional, string) the functional operation, valid values: Register, Push, Pull, Install, Remove. If not specified, it's considered as Push Operation by default.
+* `NodeConfigurationName`: (optional, string) the name of the configuration to apply.
+* `RefreshFrequencyMins`: (optional, int) pecifies how often (in minutes) DSC attempts to obtain the configuration from the pull server. 
+       If configuration on the pull server differs from the current one on the target node, it is copied to the pending store and applied.
+* `ConfigurationMode`: (optional, string) Specifies how DSC should apply the configuration. Valid values are: ApplyOnly, ApplyAndMonitor, ApplyAndAutoCorrect.
```
Comment (eshaparmar) [inline DSC/README.md]: Suggestion: Specifies DSC configuration apply mode. Valid values ...

File: DSC/dsc.py [hunk 3]
```diff
@@ -134,7 +132,12 @@ def enable():
     hutil.exit_if_enabled()
     try:
         start_omiservice()
-        mode = get_config('Mode')
+        mode_parameter = get_config('Mode')
+        if mode_parameter != '':
```
Comment (eshaparmar) [inline DSC/dsc.py]: This is a breaking change. We should communicate this to the user

File: DSC/dsc.py [hunk 4]
```diff
@@ -267,29 +288,28 @@ def install_dsc_packages():
...
-        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi', omi_major_version, omi_minor_version, omi_build)
-        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc', dsc_major_version, dsc_minor_version, dsc_build)
+        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi', omi_major_version, omi_minor_version, omi_build, omi_release)
+        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc', dsc_major_version, dsc_minor_version, dsc_build, dsc_release)
 
-def compare_pkg_version(system_package_version, major_version, minor_version, build):
-    version = re.match('(\d+).(\d+).(\d+)', system_package_version)
-    if version is not None and ((int(version.group(1)) > major_version) or (int(version.group(1)) == major_version and int(version.group(2)) > minor_version) or (int(version.group(1)) == major_version and int(version.group(2)) == minor_version and int(version.group(3)) >= build)):
+def compare_pkg_version(system_package_version, major_version, minor_version, build, release):
+    version = re.match('(\d+).(\d+).(\d+).(\d+)', system_package_version)
```
Comment (eshaparmar) [inline DSC/dsc.py]: regex pattern is same as on line 247. Should prolly be a class/script level string

File: DSC/README.md [hunk 5]
```diff
@@ -1,19 +1,19 @@
 # DSCForLinux Extension
 Allow the owner of the Azure Virtual Machines to configure the VM using Desired State Configuration (DSC) for Linux.
 
-Latest version is 2.0.
+Latest version is 3.0.
```
Comment (balukambala) [inline DSC/README.md]: Yes. Since we are changing Parameter, we can't let nodes to auto-upgrade

File: DSC/README.md (hunk 2 above)
Comment (balukambala) [inline DSC/README.md]: I copied this stmt from DSC for linux README page. Current statement helps customer to understand better.

File: DSC/dsc.py (hunk 3 above)
Comment (balukambala) [inline DSC/dsc.py]: Please give us suggestions on how we communicate to User.

**Pull request merged**